### PR TITLE
add new nsxt module nsxt_rest

### PR DESCRIPTION
**Pull request by sky-joker**
##### SUMMARY

This PR adds a module that directly accesses and executes the NSX REST API.
NSX-T is so multifunctional that the existing Ansible modules are insufficient to do everything.
This module is useful for running ranges that are not covered by existing Ansible modules.

##### PR TYPE

* New Module Pull Request

##### COMPONENT NAME

nsxt_rest

##### ADDITIONAL INFORMATION

tested with NSXT-3.0
If the branch to pull request is wrong, could you tell me the appropriate branch?

### CODE REVIEW

File: library/nsxt_rest.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+            - post
+            - put
+            - patch
+            - delete
+        default: get
+        type: str
+    src:
+        description:
+            - "The absolute path to the file containing the request body(payload) to be sent to the NSX REST API."
+            - "If this option doesn't use, use the C(content) option instead."
```
Comment (ggverma) [inline library/nsxt_rest.py:66]: typo: `doesn't use,` -> `isn't used,`

File: library/nsxt_rest.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            - "If this option is used, the C(content) option is ignored."
+        required: false
+        type: str
+    content:
+        description:
+            - "The request body(payload) to be sent to the NSX REST API."
+            - "If this option doesn't use, use the C(src) option instead."
```
Comment (ggverma) [inline library/nsxt_rest.py:73]: same

File: library/nsxt_rest.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  register: get_segment_information_result
+
+- name: delete a segment
+  nsxt_rest:
+    hostname: "{{ nsxt_hostname }}"
+    username: "{{ nsxt_username }}"
+    password: "{{ nsxt_password }}"
+    validate_certs: false
+    method: delete
+   path: /policy/api/v1/infra/segments/segment
```
Comment (ggverma) [inline library/nsxt_rest.py:114]: incorrect indent?

File: test_rest.yml [hunk 4]
```diff
@@ -0,0 +1,290 @@
+# Test code for the nsxt_rest module.
```
Comment (ggverma) [inline test_rest.yml:1]: this is great!

File: library/nsxt_rest.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                                     headers=self.headers, data=json.dumps(self.content))
+        except Exception as err:
+            self.module.fail_json(msg="nsxt rest api request error: %s, error url: %s"
+                                      % (err, self.manager_url + self.path))
+
+        if not ignore_errors:
+            self.error_code_check(info)
+
+    def delete_method(self, ignore_errors=False):
+        try:
```
Comment (ggverma) [inline library/nsxt_rest.py:220]: lines 220-227 are similar in delete, get, update methods. perhaps you could extract them to a common function to remove redundancy

**Comment by sky-joker** [comment]
Thank you @ggverma for reviewing!
I'm going to start working on fixing the pointed out.

File: library/nsxt_rest.py (hunk 5 above)
Comment (sky-joker) [reply library/nsxt_rest.py:198]: I changed the method functions to operate_nsxt function and made it more straightforward :)

File: library/nsxt_rest.py (hunk 3 above)
Comment (sky-joker) [reply library/nsxt_rest.py]: you are right, I fixed it.

**Comment by Chadd-C** [comment]
I've also used this module quite a bit to fill in gaps for things that don't have a specific module yet... Any chance this can might pulled in?

**Review by ggverma — APPROVED** [review]

**Comment by sky-joker** [comment]
@madhukark @ggverma 
Thank you for merging!

**Pull request merged**